model/metrics/messaging-metrics.yaml

```diff
@@ -0,0 +1,80 @@
+    brief: "Common messaging metrics attributes."
+    extends: attributes.messaging.common.minimal
+    attributes:
+      - ref: messaging.system
+        requirement_level: required
+      - ref: messaging.destination.partition.id
+      - ref: messaging.destination.name
+        requirement_level:
+          conditionally_required: if and only if `messaging.destination.name` is known to have low cardinality. Otherwise, `messaging.destination.template` MAY be populated.
+      - ref: messaging.destination.template
+        requirement_level:
+          conditionally_required: if available.
+
+  # durations
+  - id: metric.messaging.client.request.duration
+    type: metric
+    metric_name: messaging.client.request.duration
+    brief: "Duration of messaging operation initiated by a producer or consumer."
+    note: >
+      This metric SHOULD NOT be used to report processing duration - processing duration is reported in `messaging.process.duration` metric.
+    stability: experimental
+    instrument: histogram
+    unit: "s"
+    extends: metric.messaging.attributes
+    attributes:
+      - ref: messaging.operation.type
+        requirement_level:
+          conditionally_required: If applicable.
+      - ref: messaging.operation.name
+        examples: ["send", "receive", "ack"]
+
+  - id: metric.messaging.process.duration
+    type: metric
+    metric_name: messaging.process.duration
+    brief: "Duration of processing operation."
+    note: >
+      This metric MUST be reported for operations with `messaging.operation.type` that matches `process`.
+    stability: experimental
+    instrument: histogram
+    unit: "s"
+    extends: metric.messaging.attributes
+    attributes:
+      - ref: messaging.operation.name
+        examples: ["process", "consume", "handle"]
+
+  # counters
+  - id: metric.messaging.client.published.messages
+    type: metric
+    metric_name: messaging.client.published.messages
+    brief: "Number of messages producer attempted to publish to the broker."
+    note: >
+      This metric MUST NOT count messages that were created, but aren't being published yet.
+    stability: experimental
+    instrument: counter
+    unit: "{message}"
+    extends: metric.messaging.attributes
+    attributes:
+      - ref: messaging.operation.name
+        examples: ["send", "schedule", "enqueue"]
+
+  - id: metric.messaging.client.consumed.messages
+    type: metric
+    metric_name: messaging.client.consumed.messages
+    brief: "Number of messages that were delivered to the application."
+    note: >
+      Records the number of messages pulled from the broker or number of messages dispatched to the application in push-based scenarios.
+
+      The metric MUST be reported once per message delivery. For example, if receiving and processing operations are both instrumented for a single message delivery, the number of received messages
+      is reported when then message is received and not reported for the processing operation.
```

Comment:
maybe MUST -> SHOULD?

this could be tricky to implement / coordinate, e.g. when we instrument Spring messaging to capture "process" operations, and Spring uses Kafka polling under the hood which is also instrumented to capture "receive" operation

Reply:
> maybe MUST -> SHOULD?

done 👍 

In the spring-kafka case, Spring integration should not report the number of consumed messages since kafka instrumentation does. But I agree it's could be impossible to know in some cases.